PyInstaller/archive/writers.py

```diff
@@ -222,7 +224,10 @@ def _add_from_table_of_contents(self, toc):
           entry[2] is a flag for it's storage format (True or 1 if compressed)
           entry[3] is the entry's type code.
         """
-        for toc_entry in toc:
+        toc2 = copy.copy(toc)
+        toc2.sort(key=itemgetter(0), reverse=False)  # in place sort
```

Comment:
A `TOC` is a list and lists are sorted by their first item anyway. No need for all this code. `toc.sort()` would to the same.